gst/nnstreamer/tensor_query/tensor_query_serversink.c

```diff
@@ -110,7 +177,134 @@ static void
 gst_tensor_query_serversink_get_property (GObject * object, guint prop_id,
     GValue * value, GParamSpec * pspec)
 {
-  return;
+  GstTensorQueryServerSink *serversink = GST_TENSOR_QUERY_SERVERSINK (object);
+
+  switch (prop_id) {
+    case PROP_HOST:
+      g_value_set_string (value, serversink->host);
+      break;
+    case PROP_PORT:
+      g_value_set_int (value, serversink->port);
+      break;
+    case PROP_PROTOCOL:
+      g_value_set_int (value, serversink->protocol);
+      break;
+    case PROP_TIMEOUT:
+      g_value_set_int (value, serversink->timeout);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief start rocessing of query_serversrc
+ */
+static gboolean
+gst_tensor_query_serversink_start (GstBaseSink * bsink)
+{
+  GstTensorQueryServerSink *sink = GST_TENSOR_QUERY_SERVERSINK (bsink);
+  GstTensorsConfig config;
+
+  gst_tensors_config_from_peer (bsink->sinkpad, &config, NULL);
+  config.rate_n = 0;
+  config.rate_d = 1;
+  if (!gst_tensors_config_validate (&config)) {
+    nns_loge ("Invalid tensors config from peer");
+    return FALSE;
+  }
+  gst_tensor_query_server_set_sink_config (&config);
+  gst_tensors_config_free (&config);
+
+  if (!sink->server_data) {
+    nns_loge ("Server_data is NULL");
+    return FALSE;
+  }
+
+  if (nnstreamer_query_server_init (sink->server_data, sink->protocol,
+          sink->host, sink->port) != 0) {
+    nns_loge ("Failed to setup server");
+    return FALSE;
+  }
+  return TRUE;
+}
+
+/**
+ * @brief stop processing of query_serversrc
+ */
+static gboolean
+gst_tensor_query_serversink_stop (GstBaseSink * bsink)
+{
+  GstTensorQueryServerSink *sink = GST_TENSOR_QUERY_SERVERSINK (bsink);
+  nnstreamer_query_server_data_free (sink->server_data);
+  return TRUE;
 }
 
-#pragma GCC diagnostic pop
+/**
+ * @brief render buffer, send buffer to client 
+ */
+static GstFlowReturn
+gst_tensor_query_serversink_render (GstBaseSink * bsink, GstBuffer * buf)
+{
+  GstTensorQueryServerSink *sink = GST_TENSOR_QUERY_SERVERSINK (bsink);
+  GstMetaQuery *meta_query;
+  TensorQueryCommandData cmd_data;
+  query_connection_handle conn;
+  GstMemory *mem;
+  GstMapInfo map;
+  gchar *host;
+  guint32 i, num_mems;
+
+  meta_query = gst_buffer_get_meta_query (buf);
+  while (TRUE) {
+    conn = nnstreamer_query_server_accept (sink->server_data);
+    host = nnstreamer_query_connection_get_host (conn);
+    if (g_str_equal (host, meta_query->host)) {
```

Comment:
If multiple connections are made from clients with the same ip, port information of the client may be required.
But..... in this case, how can we distinguish between two clients?

Reply:
Concurrent query from same IP is not possible for now because server(sink) has only IP address to identify..

Reply:
Let's implement this in the next PR.